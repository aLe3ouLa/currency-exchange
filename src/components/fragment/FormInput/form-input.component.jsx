import React from "react";

import { Input, Sign } from "./form-input.styles";

const FormInput = ({
  name,
  value,
  onChange,
  disabled,
  isFromValue,
  isZero,
}) => {
  return (
    <React.Fragment>
      <Sign>{value ? (isFromValue ? `-` : `+`) : ""}</Sign>
      <Input
        isZero={isZero}
        name={name}
        value={value}
        placeholder="0"
        onChange={onChange}
        disabled={disabled}
        autoComplete="off"
      />
    </React.Fragment>
  );
};
export default FormInput;
