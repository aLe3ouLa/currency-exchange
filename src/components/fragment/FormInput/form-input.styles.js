import styled, { css } from "styled-components";

export const Input = styled.input`
  text-align: right;
  border: none;
  outline: none;
  background: transparent;
  font-size: 3rem;
  display: flex;
  width: 70%;
  ${({ isZero }) =>
    isZero
      ? css`
          color: ${({ theme }) => theme.color.gray};
        `
      : css`
          color: ${({ theme }) => theme.color.text};
        `};
`;

export const Sign = styled.span`
  font-size: 3rem;
`;
