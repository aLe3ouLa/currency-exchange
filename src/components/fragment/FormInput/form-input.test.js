import React from "react";
import ReactDOM from "react-dom";
import FormInput from "./form-input.component";

import { cleanup, render, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme, Defaults } from "../../../styles";

describe("Form Input", () => {
  afterEach(cleanup);
  const props = {
    name: "value",
    value: "",
    onChange: jest.fn(),
    disabled: false,
    isFromValue: true,
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <FormInput {...props} />
        </React.Fragment>
      </ThemeProvider>,
      div
    );
  });

  it("has a valid snapshot ", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <FormInput {...props} />
        </React.Fragment>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should call handleChange method when input changes", () => {
    const handleChange = jest.fn();
    const { getByLabelText } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <FormInput {...props} onChange={handleChange} />
        </React.Fragment>
      </ThemeProvider>
    );

    fireEvent.change(getByLabelText("form-input"), {
      target: { value: 1 },
    });

    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it("should not accept type other than number", () => {
    let value;
    const handleChange = (ev) => {
      ev.preventDefault();
      value = ev.target.value;
    };
    const { getByLabelText } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <FormInput {...props} onChange={handleChange} />
        </React.Fragment>
      </ThemeProvider>
    );

    fireEvent.change(getByLabelText("form-input"), {
      target: { value: "a" },
    });

    expect(value).toBe(undefined); //default value
  });

  it("should not accept type other than number", () => {
    let value;
    const handleChange = (ev) => {
      ev.preventDefault();
      value = ev.target.value;
    };
    const { getByLabelText } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <FormInput {...props} onChange={handleChange} />
        </React.Fragment>
      </ThemeProvider>
    );

    const input = getByLabelText("form-input");

    fireEvent.change(input, { target: { value: "23.0" } });
    expect(value).toBe("23.0");
  });
});
