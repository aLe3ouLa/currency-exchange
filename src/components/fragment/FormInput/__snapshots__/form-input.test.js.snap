// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Input has a valid snapshot  1`] = `
<DocumentFragment>
  <span
    class="sc-AxirZ kBcKOJ"
  />
  <input
    aria-label="form-input"
    autocomplete="off"
    class="sc-AxjAm ibqwxn"
    name="value"
    placeholder="0"
    type="number"
    value=""
  />
</DocumentFragment>
`;
